vms/platformvm/vm_test.go

```diff
@@ -2118,3 +2118,68 @@ func TestRemovePermissionedValidatorDuringAddPending(t *testing.T) {
 	_, err = vm.state.GetPendingValidator(createSubnetTx.ID(), ids.NodeID(id))
 	require.ErrorIs(err, database.ErrNotFound)
 }
+
+func TestTransferSubnetOwnershipTx(t *testing.T) {
+	require := require.New(t)
+	vm, _, _ := defaultVM(t)
+	vm.ctx.Lock.Lock()
+	defer func() {
+		require.NoError(vm.Shutdown(context.Background()))
+		vm.ctx.Lock.Unlock()
+	}()
+
+	// Create a subnet
+	createSubnetTx, err := vm.txBuilder.NewCreateSubnetTx(
+		1,
+		[]ids.ShortID{keys[0].PublicKey().Address()},
+		[]*secp256k1.PrivateKey{keys[0]},
+		keys[0].Address(),
+	)
+	require.NoError(err)
+	subnetID := createSubnetTx.ID()
+
+	require.NoError(vm.Builder.AddUnverifiedTx(createSubnetTx))
+	createSubnetBlock, err := vm.Builder.BuildBlock(context.Background())
```

Comment:
nit: should we add a type assertion to verify createSubnetBlock is a standard block? Should we check it contains the subnet creation tx? It may feel like overengineering but this kind of checks really simplify maintenance. It happened already in the past that UT were green when they should have failed